src/include/txn_inline.h

```diff
@@ -706,6 +706,8 @@ __wt_txn_modify(WT_SESSION_IMPL *session, WT_UPDATE *upd)
     WT_ASSERT(session, !WT_IS_HS((S2BT(session))->dhandle));
 
     upd->txnid = session->txn->id;
+    upd->prepared_id = txn->prepared_id;
```

Comment:
The change of moving the prepared timestamp and prepare_id should be done in `__txn_apply_prepare_state_update` function. Please move the code there.